compiler/ir/backend.js/src/org/jetbrains/kotlin/ir/backend/js/transformers/irToJs/JsIntrinsicTransformers.kt

```diff
@@ -319,5 +322,27 @@ private inline fun MutableMap<IrSymbol, IrCallTransformer>.withTranslatedArgs(
     function: IrSimpleFunctionSymbol,
     crossinline t: (List<JsExpression>) -> JsExpression
 ) {
-    put(function) { call, context -> t(translateCallArguments(call, context)) }
+    put(function) { call, context ->
+        runCatching {
+            t(translateCallArguments(call, context))
+        }.getOrElse {
+            doError(context.createErrorContextInfo(), cause = it)
+        }
+    }
+}
+
+private fun IrCallTransformer.wrappedWithErrorHandler(): IrCallTransformer = { call, context ->
+    runCatching {
+        this@wrappedWithErrorHandler(call, context)
+    }.getOrElse {
+        doError(context.createErrorContextInfo(), cause = it)
+    }
 }
+
+private fun doError(
+    vararg args: Any,
```

Comment:
I'm not a fan of having multiple different purposes for the same code, e.g. here with `Any` in the signature and a default argument. I'd rather use `error` with specialized error message wherever needed.

Reply:
I am agree with you, but currently it is easier way to log environment of error and put it in error. I may rename function, but currently do not understand what should I do with args here